src/main/kotlin/com/autonomousapps/tasks/classAnalysisTasks.kt

```diff
@@ -119,31 +124,33 @@ abstract class JarAnalysisTask @Inject constructor(
 
 interface JarAnalysisParameters : WorkParameters {
   val variantFiles: SetProperty<VariantFile>
-  var jar: File
-  var kaptJavaSource: Set<File>
-  var layouts: Set<File>
-  var testFiles: Set<File>
-  var report: File
-  var reportPretty: File
+  var jar: RegularFileProperty
+  var kaptJavaSource: FileCollection
+  var layouts: FileCollection
+  var testJavaClassesDir: DirectoryProperty
+  var testKotlinClassesDir: DirectoryProperty
+  var report: RegularFileProperty
+  var reportPretty: RegularFileProperty
 }
 
 abstract class JarAnalysisWorkAction : WorkAction<JarAnalysisParameters> {
 
   private val logger = getLogger<JarAnalysisTask>()
 
   override fun execute() {
+    val testFiles = getTestFiles(parameters.testJavaClassesDir, parameters.testKotlinClassesDir)
     val classNames = JarReader(
       variantFiles = parameters.variantFiles.get(),
-      jarFile = parameters.jar,
-      layouts = parameters.layouts,
-      testFiles = parameters.testFiles,
-      kaptJavaSource = parameters.kaptJavaSource
+      jarFile = parameters.jar.get().asFile,
+      layouts = parameters.layouts.files,
+      testFiles = testFiles,
+      kaptJavaSource = parameters.kaptJavaSource.files
     ).analyze()
 
-    parameters.report.writeText(classNames.toJson())
-    parameters.reportPretty.writeText(classNames.toPrettyString())
+    parameters.report.getAndDelete().writeText(classNames.toJson())
```

Comment:
as noted above, I believe it is important to delete at the very beginning of the execution. I would also accept doing it at the beginning of the worker action (for simple tasks that only delegate to workers and doing nothing else). But I feel strongly it shouldn't happen at the _end_.